### ISSUE TITLE
Jupyter notebook for Parameter Sweep tutorial occasional cell timeout failures on Linux/Python 3.8

### ISSUE THREAD

**Issue by lbianchi-lbl** [watertap-org/watertap#1312]
![image](https://github.com/watertap-org/watertap/assets/48035537/bdf8416f-61ea-411e-9839-28700d6647f3)

**Comment by adam-a-a** [comment]
Just noting that I am seeing a similar issue in checks for PR #1244 : 
![image](https://github.com/watertap-org/watertap/assets/22691844/fbe2254c-2787-46d1-800e-faee7925f1a4)

**Comment by lbianchi-lbl** [comment]
Based on my testing in #1314, the failures occur on Linux with Python 3.8 up for (up to) about half of the runs:

![image](https://github.com/watertap-org/watertap/assets/48035537/03e6273f-fb43-4101-bb4b-dba26f89b718)

**Comment by bknueven** [comment]
It looks like the demo uses the "ConcurrentFutures" backend: https://github.com/watertap-org/watertap/blob/a3eb07794c9852f270d4fdde1b94d312e0708fa3/tutorials/parameter_sweep_demo.ipynb#L224.

**Comment by bknueven** [comment]
Given that Python 3.9 seems to be working, here are the release notes for Python 3.9 about `concurrent.futures`: https://docs.python.org/3/whatsnew/3.9.html#concurrent-futures

**Comment by bknueven** [comment]
Based on our experience with #1211 it's almost certainly the case that the deadlock is occurring in the `gather` method:
https://github.com/watertap-org/watertap/blob/main/watertap/tools/parallel/concurrent_futures_parallel_manager.py#L104-L115

**Comment by bknueven** [comment]
As a hack, the multiprocessing parallel back end seems to work fine: https://github.com/watertap-org/watertap/actions/runs/8080874093/job/22078190530

**Comment by bknueven** [comment]
Given the available evidence I am leaning towards this being a bug in Python 3.8. Python 3.8 hits EOL in October 2024.

How much resources should we devote to fixing this issue, when a work-around (use `MultiProcessing`) is readily available? It also raises the question of the utility of having both a `multiprocessing` parallel back end and a `concurrent.futures` parallel back end. If both are included with Python, why maintain both?

**Comment by lbianchi-lbl** [comment]
Thanks for running the checks @bknueven. Personally, I'd be more than happy to just consider this a Python 3.8 limitation, and not do anything until the time when we drop support for Python 3.8 beyond excluding the notebook from being tested in Python 3.8 CI jobs.

**Comment by bknueven** [comment]
Okay, I will implement the work around for Python 3.8 so this doesn't hold up further jobs.

**Comment by lbianchi-lbl** [comment]
Just to be clear, as far as I'm concerned the "workaround" can be

```yaml
if: matrix.python-version == '3.8' && matrix.os == 'linux'`
run: rm tutorials/parameter_sweep_demo.ipynb
```

in the GHA job where notebooks are tested, meaning I don't think we need to come up with a "proper" fix.

**Comment by bknueven** [comment]
> Just to be clear, as far as I'm concerned the "workaround" can be
> 
> ```yaml
> if: matrix.python-version == '3.8' && matrix.os == 'linux'`
> run: rm tutorials/parameter_sweep_demo.ipynb
> ```
> 
> in the GHA job where notebooks are tested, meaning I don't think we need to come up with a "proper" fix.

I already had a work-around slightly more complex than this implemented (#1316) -- which should help any users of Python 3.8 not hit this same issue.
